### ISSUE TITLE
Not fully backward compatible with scrapbook

### ISSUE THREAD

**Issue by nik-gr** [vctfence/scrapbee#79]
You cannot edit with scrapbook pages saved using scrapbee which is a big drawback. Will that be fixed in a future version? Most likely that is because scrapbee uses more number digits in the folders IDs. The scrapbook approach uses a timestamp up to the second, why increase it to more than that? The only case when someone may endup saving more than one pages in the same second is when choose to save all tabs of a window but even then it can be managed. If a folder with the same ID exists check for a new ID increased by 1 compare to the existed one until find one that has not yet been created.

Also there is no choice to choose filetypes with direct links in the page that you can save, for example pdf links.
And if a pdf is opened with the inline pdf reader of firefox, it cannot be saved, there is an error message about not being allowed to do it.
Are there any plans to implement these?

**Comment by touwys** [comment]
I concur with the original poster. I am still maintaining huge scrapbooks created with Scrapbook, Scrapbook Plus, and Scrapbook X — all 3 of which are suitably compatible. Therefore, there is a great case to be made out for Scrapbee to follow suit. 

In addition, it is essential to be able to save PDF and Easy Read links as well as the commentary sections of all pages — all of which are core features for this type of app.

**Comment by vctfence** [comment]
Hi, that's because the origin id format  is too weak to fulfill the node mover tool -- if you generate ids rapidly(for example, 10 ids in 1 second), you'll get some duplicate ids.

**Comment by nik-gr** [comment]
I haven't used the node mover tool. If I am not mistaken is to move 'scrapped' files right? Are you saying that it changes IDs when moving them? But why, I mean as metadata it makes more sense to keep the date when you downloaded a file not the date you moved it

**Comment by nik-gr** [comment]
sorry I'm not sure I follow you, ids are used for all the items in the rdf (pages,folders,bookmars) right ? it wouldn't make any difference if you had changed the format completely: legacy scrapbook like would still be able to search the saved pages but not to edit them as it is now with this format. It is not usable if you want to access it with legacy scrapbook

I didn't try node mover so I'm not sure what it does, if it is like the legacy export/import feature. In any case I don't understand the need for new ids if you move it to a different scrapbook folder and I don't see why someone would want to move it with new ids in the same scrapbook folder.

Can you also please answer my other question? is it in your plans to provide a feature where the user can choose filetypes to be saved if they are as links in the page to be saved (like pdf/php/image links)?

**Comment by vctfence** [comment]
Do you mean just to save a media file, such as a image? I'll try to provide "save media under the cursor" context menu item in the future.

**Comment by nik-gr** [comment]
Hi, I'm talking about something similar to what scrapbook provided:  
The choice to give filetypes by file extenisons like ".mp4",".pdf", ".jpg" etc and when the page is saved all links with those specific extensions will be downloaded as well, in the same folder ID and in the saved page those links will now direct to the downloaded files. 
Also  if I remember correctly, a welcome addition in the special case when you added ".php" or similar extensions, it was downloading the documents/files that the specific link would have return if it was clicked (instead of the php code)

**Comment by vctfence** [comment]
Right, Scrapbee does not save contents linked to a page so far, mainly, Scrapbee is young and there's a risk to get a huge result. This maybe changed in the future.
